python/pfs/drp/stella/buildFiberProfiles.py

```diff
@@ -77,31 +82,40 @@ def run(self, exposure, detectorMap=None, pfsConfig=None):
             Callable for each fiber (indexed by fiberId) that provides the
             center of the trace as a function of row.
         """
-        if self.config.doBlindFind or detectorMap is None or pfsConfig is None:
+        if self.config.doBlindFind or detectorMap is None:
             convolved = self.convolveImage(exposure.maskedImage)
-            traces = self.findPeaks(convolved)
-            self.log.debug("Found %d peaks", sum([len(pp) for pp in traces]))
-            traces = self.prunePeaks(traces)
-            self.log.debug("Pruned to %d peaks", sum([len(pp) for pp in traces]))
-            traces = self.associatePeaks(traces, convolved.getWidth())
+            peaks = self.findPeaks(convolved)
+            self.log.debug("Found %d peaks", sum([len(pp) for pp in peaks]))
+            peaks = self.prunePeaks(peaks)
+            self.log.debug("Pruned to %d peaks", sum([len(pp) for pp in peaks]))
+            traces = self.associatePeaks(peaks, convolved.getWidth())
             self.log.debug("Associated into %d traces", len(traces))
             traces = self.pruneTraces(traces, exposure.getHeight())
             self.log.debug("Pruned to %d traces", len(traces))
+            traces = {ii: tt for ii, tt in enumerate(traces)}
         else:
             traces = self.generateTraces(detectorMap, pfsConfig)
-        profiles = FiberProfileSet.makeEmpty(exposure.getInfo().getVisitInfo(), exposure.getMetadata())
-        centers = {}
-        for ii, tt in enumerate(traces):
+
+        # Centroid the traces.
+        # If we don't have a detectorMap, we will fit a functional form to get the centers for the profiles.
+        # If we do have a detectorMap, this will be used to adjust the detectorMap's centers.
```

Comment:
I would expand this to say something along the lines of  'If we do have a detectorMap, the traces will be used to adjust the detectorMap's centers to yield a more accurate functional form for the profile centers.'